ftd_api/bulk_tool.py

```diff
@@ -294,43 +244,61 @@ def do_upload_import_file(self, file_name):
         else:
             file_name_without_path = os.path.split(file_name)[1]
         with open(file_name) as upload_filehandle:
-            return self.do_upload_import_dict_list(json.loads(upload_filehandle.read()), upload_file_name_without_path=file_name_without_path)
+            return self._do_upload_import_dict_list(json.loads(upload_filehandle.read()), upload_file_name_without_path=file_name_without_path)
 
-    def extract_config_file_from_export(self, export_zip_file, dest_directory):
+    def _extract_config_file_from_export(self, export_zip_file, dest_directory, export_type=None):
         """
         This method will extract the configuration file from the zip file
 
         Parameters:
 
         export_zip_file -- This is the input zip file
         dest_directory -- This is the directory to extract the contents into
+        export_type -- Optional specifies the type of export being done (FULL_EXPORT, PENDING_CHANGE_EXPORT or PARTIAL_EXPORT)
 
         Return value is the full path to the extracted config file
         """
         with zipfile.ZipFile(export_zip_file, 'r') as zip_ref:
             zip_ref.extractall(dest_directory) #test for file name
-        config_file_name = dest_directory + '/' + 'full_config.txt'
-        if not os.path.isfile(config_file_name):
-            config_file_name = dest_directory + '/' + 'pending_change_config.txt'
-            if not os.path.isfile(config_file_name):
+        
+        # Possible types: FULL_EXPORT, PENDING_CHANGE_EXPORT and PARTIAL_EXPORT
+        if export_type is not None:

```

Comment:
FYI I did this change so if you run the tool on the same directory with different export flags it will know which file to read.  Without the hint it would just scan and take the first match which could have been incorrect.